### ISSUE TITLE
Bug: Validation rules `integer` and `numeric` accepts boolean

### ISSUE THREAD

**Issue by shishamo** [codeigniter4/CodeIgniter4#5374]
### PHP Version

7.4

### CodeIgniter4 Version

4.1.5

### CodeIgniter4 Installation Method

Composer (as dependency to an existing project)

### Which operating systems have you tested for this bug?

macOS

### Which server did you use?

apache

### Database

MariaDB 10.2

### What happened?

The validation rules "integer" and "numeric" accepts boolean true&false and return true

### Steps to Reproduce

```php
$data['numeric'] = false;
$data['integer'] = true;

$validation = service('validation');
$validation->setRules([
    'numeric' => 'numeric',
    'integer' => 'integer'
]);

d($validation->run($data));
```

### Expected Output

The validation should fail and return false?

**Comment by kenjis** [comment]
Thank you for reporting.

These might not be bugs. Because the rule `integer` and `numeric` only accept `string`.

https://github.com/codeigniter4/CodeIgniter4/blob/3b1e98473833571feeba663928bdb8d0ea4b886a/system/Validation/FormatRules.php#L122
https://github.com/codeigniter4/CodeIgniter4/blob/3b1e98473833571feeba663928bdb8d0ea4b886a/system/Validation/FormatRules.php#L146

So the bool `true` converted to `1` after all and the validation returns `true`.
But I'm not sure these behaviors are good.

These validation rules seems to be only for validating data from web site users like `$_POST`.

**Comment by kenjis** [comment]
@shishamo What data do you validate?

**Comment by shishamo** [comment]
@kenjis 
Thanks for your answer.
I am creating an API and need to validate json data post by 'Content-Type: application/json'

**Comment by kenjis** [comment]
@shishamo Please try.
```
$validation->setRules([
    'numeric' => 'is_numeric',
    'integer' => 'is_int'
]);
```

**Comment by kenjis** [comment]
> I am creating an API and need to validate json data post by 'Content-Type: application/json'

It seems validating only string is out-of-dated.

**Comment by shishamo** [comment]
@kenjis 
```
$validation->setRules([
    'numeric' => 'is_numeric',
    'integer' => 'is_int'
]);
```

Looks like working but accepts array as below
```
[]
[555]
```
> It seems validating only string is out-of-dated.

Maybe for a next update could be nice? ^^

**Comment by kenjis** [comment]
It is natural that you validate JSON data with the Validation class,
and the results may not be correct, it is not good for security.

I would like to fix the problems as bug fixes. 
So I created PR #5375.
But it has big breaking changes, so I don't know it is contained in the next update.

**Comment by shishamo** [comment]
Thank you for your support.

Looking forward for the new updates.

**Comment by shishamo** [comment]
Hi guys
Thanks you for the issue support.
I have updated to the version 4.1.6 and the validation works for [5] or the boolean true.
Sorry to reopen this tread,  but it still doesn't handle the boolean false or muti-dimensional array for ex: [[5]]

**Comment by kenjis** [comment]
@shishamo This issue is not fixed in 4.1.6 yet, and not closed.
This would be fixed in #5445.

**Comment by shishamo** [comment]
I see @kenjis 
Thank you very much
